src/cryptsetup/cryptsetup-tokens/cryptsetup-token-systemd-fido2.c

```diff
@@ -39,7 +39,7 @@ _public_ int cryptsetup_token_open_pin(
 
         /* This must not fail at this moment (internal error) */
         r = crypt_token_json_get(cd, token, &json);
-        assert(token == r);
+        assert_se(token == r);
```

Comment:
Could you elaborate this change?

Reply:
Ah, I forgot to add the error to the commit message, my apologies:

```
2022-05-21T15:25:06.8452819Z FAILED: libcryptsetup-token-systemd-fido2.so.p/src_cryptsetup_cryptsetup-tokens_cryptsetup-token-systemd-fido2.c.o 
2022-05-21T15:25:06.8457199Z clang-13 -Ilibcryptsetup-token-systemd-fido2.so.p -I. -I.. -Isrc/basic -I../src/basic -Isrc/fundamental -I../src/fundamental -Isrc/systemd -I../src/systemd -I../src/libsystemd/sd-bus -I../src/libsystemd/sd-device -I../src/libsystemd/sd-event -I../src/libsystemd/sd-hwdb -I../src/libsystemd/sd-id128 -I../src/libsystemd/sd-journal -I../src/libsystemd/sd-netlink -I../src/libsystemd/sd-network -I../src/libsystemd/sd-resolve -Isrc/shared -I../src/shared -fcolor-diagnostics -DNDEBUG -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -Werror -std=gnu11 -O0 -g -Wno-missing-field-initializers -Wno-unused-parameter -Wdate-time -Wendif-labels -Werror=format=2 -Werror=implicit-function-declaration -Werror=incompatible-pointer-types -Werror=int-conversion -Werror=overflow -Werror=override-init -Werror=return-type -Werror=shift-count-overflow -Werror=undef -Wfloat-equal -Winit-self -Wmissing-include-dirs -Wmissing-noreturn -Wnested-externs -Wold-style-definition -Wpointer-arith -Wredundant-decls -Wshadow -Wstrict-aliasing=2 -Wstrict-prototypes -Wunused-function -Wwrite-strings '-Wno-error=#warnings' -Wno-string-plus-int -Wno-typedef-redefinition -Wno-gnu-variable-sized-type-not-at-end -Werror=missing-declarations -Werror=missing-prototypes -fdiagnostics-show-option -fno-common -fno-strict-aliasing -fstack-protector -fstack-protector-strong -fvisibility=hidden --param=ssp-buffer-size=4 -Werror=shadow -include config.h -Werror -fPIC -MD -MQ libcryptsetup-token-systemd-fido2.so.p/src_cryptsetup_cryptsetup-tokens_cryptsetup-token-systemd-fido2.c.o -MF libcryptsetup-token-systemd-fido2.so.p/src_cryptsetup_cryptsetup-tokens_cryptsetup-token-systemd-fido2.c.o.d -o libcryptsetup-token-systemd-fido2.so.p/src_cryptsetup_cryptsetup-tokens_cryptsetup-token-systemd-fido2.c.o -c ../src/cryptsetup/cryptsetup-tokens/cryptsetup-token-systemd-fido2.c
2022-05-21T15:25:06.8460369Z ../src/cryptsetup/cryptsetup-tokens/cryptsetup-token-systemd-fido2.c:33:13: error: variable 'r' set but not used [-Werror,-Wunused-but-set-variable]
2022-05-21T15:25:06.8460944Z         int r;
2022-05-21T15:25:06.8461146Z             ^
2022-05-21T15:25:06.8461321Z 1 error generated.
2022-05-21T15:25:06.9587264Z [516/2143] clang-13 -Ilibcryptsetup-token-systemd-pkcs11.so.p -I. -I.. -Isrc/basic -I../src/basic -Isrc/fundamental -I../src/fundamental -Isrc/systemd -I../src/systemd -I../src/libsystemd/sd-bus -I../src/libsystemd/sd-device -I../src/libsystemd/sd-event -I../src/libsystemd/sd-hwdb -I../src/libsystemd/sd-id128 -I../src/libsystemd/sd-journal -I../src/libsystemd/sd-netlink -I../src/libsystemd/sd-network -I../src/libsystemd/sd-resolve -Isrc/shared -I../src/shared -I/usr/include/p11-kit-1 -fcolor-diagnostics -DNDEBUG -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -Werror -std=gnu11 -O0 -g -Wno-missing-field-initializers -Wno-unused-parameter -Wdate-time -Wendif-labels -Werror=format=2 -Werror=implicit-function-declaration -Werror=incompatible-pointer-types -Werror=int-conversion -Werror=overflow -Werror=override-init -Werror=return-type -Werror=shift-count-overflow -Werror=undef -Wfloat-equal -Winit-self -Wmissing-include-dirs -Wmissing-noreturn -Wnested-externs -Wold-style-definition -Wpointer-arith -Wredundant-decls -Wshadow -Wstrict-aliasing=2 -Wstrict-prototypes -Wunused-function -Wwrite-strings '-Wno-error=#warnings' -Wno-string-plus-int -Wno-typedef-redefinition -Wno-gnu-variable-sized-type-not-at-end -Werror=missing-declarations -Werror=missing-prototypes -fdiagnostics-show-option -fno-common -fno-strict-aliasing -fstack-protector -fstack-protector-strong -fvisibility=hidden --param=ssp-buffer-size=4 -Werror=shadow -include config.h -Werror -fPIC -MD -MQ libcryptsetup-token-systemd-pkcs11.so.p/src_cryptsetup_cryptsetup-tokens_cryptsetup-token-systemd-pkcs11.c.o -MF libcryptsetup-token-systemd-pkcs11.so.p/src_cryptsetup_cryptsetup-tokens_cryptsetup-token-systemd-pkcs11.c.o.d -o libcryptsetup-token-systemd-pkcs11.so.p/src_cryptsetup_cryptsetup-tokens_cryptsetup-token-systemd-pkcs11.c.o -c ../src/cryptsetup/cryptsetup-tokens/cryptsetup-token-systemd-pkcs11.c
2022-05-21T15:25:06.9590826Z FAILED: libcryptsetup-token-systemd-pkcs11.so.p/src_cryptsetup_cryptsetup-tokens_cryptsetup-token-systemd-pkcs11.c.o 
2022-05-21T15:25:06.9595399Z clang-13 -Ilibcryptsetup-token-systemd-pkcs11.so.p -I. -I.. -Isrc/basic -I../src/basic -Isrc/fundamental -I../src/fundamental -Isrc/systemd -I../src/systemd -I../src/libsystemd/sd-bus -I../src/libsystemd/sd-device -I../src/libsystemd/sd-event -I../src/libsystemd/sd-hwdb -I../src/libsystemd/sd-id128 -I../src/libsystemd/sd-journal -I../src/libsystemd/sd-netlink -I../src/libsystemd/sd-network -I../src/libsystemd/sd-resolve -Isrc/shared -I../src/shared -I/usr/include/p11-kit-1 -fcolor-diagnostics -DNDEBUG -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -Werror -std=gnu11 -O0 -g -Wno-missing-field-initializers -Wno-unused-parameter -Wdate-time -Wendif-labels -Werror=format=2 -Werror=implicit-function-declaration -Werror=incompatible-pointer-types -Werror=int-conversion -Werror=overflow -Werror=override-init -Werror=return-type -Werror=shift-count-overflow -Werror=undef -Wfloat-equal -Winit-self -Wmissing-include-dirs -Wmissing-noreturn -Wnested-externs -Wold-style-definition -Wpointer-arith -Wredundant-decls -Wshadow -Wstrict-aliasing=2 -Wstrict-prototypes -Wunused-function -Wwrite-strings '-Wno-error=#warnings' -Wno-string-plus-int -Wno-typedef-redefinition -Wno-gnu-variable-sized-type-not-at-end -Werror=missing-declarations -Werror=missing-prototypes -fdiagnostics-show-option -fno-common -fno-strict-aliasing -fstack-protector -fstack-protector-strong -fvisibility=hidden --param=ssp-buffer-size=4 -Werror=shadow -include config.h -Werror -fPIC -MD -MQ libcryptsetup-token-systemd-pkcs11.so.p/src_cryptsetup_cryptsetup-tokens_cryptsetup-token-systemd-pkcs11.c.o -MF libcryptsetup-token-systemd-pkcs11.so.p/src_cryptsetup_cryptsetup-tokens_cryptsetup-token-systemd-pkcs11.c.o.d -o libcryptsetup-token-systemd-pkcs11.so.p/src_cryptsetup_cryptsetup-tokens_cryptsetup-token-systemd-pkcs11.c.o -c ../src/cryptsetup/cryptsetup-tokens/cryptsetup-token-systemd-pkcs11.c
2022-05-21T15:25:06.9646280Z ../src/cryptsetup/cryptsetup-tokens/cryptsetup-token-systemd-pkcs11.c:34:13: error: variable 'r' set but not used [-Werror,-Wunused-but-set-variable]
2022-05-21T15:25:06.9646679Z         int r;
2022-05-21T15:25:06.9646856Z             ^
2022-05-21T15:25:06.9647035Z 1 error generated.
2022-05-21T15:25:07.0143564Z [517/2143] /home/runner/.local/bin/meson --internal exe --capture src/core/bpf/socket_bind/socket-bind.skel.h -- /usr/sbin/bpftool gen skeleton src/core/bpf/socket_bind/socket-bind.bpf.o
2022-05-21T15:25:07.0144414Z ninja: build stopped: subcommand failed.
2022-05-21T15:25:07.0656402Z + fatal ''\''meson compile'\'' failed with -Db_ndebug=true'
```

Reply:
I see. Then, could you add a short description why we need to use `assert_se()` here?

Reply:
Will do, together with updating the commit message.